src/BalanceGuard.ts

```diff
@@ -117,29 +117,26 @@ export default class BalanceGuard {
       const approve = async ({ spenderAddress, targetAmount }: ApproveFix) => {
         const bnTargetAmount = new BigNumber(targetAmount);
         const unsignedApproveTx = await tokenContract
-          .populateTransaction
           .approve(
             spenderAddress,
             bnTargetAmount.isZero()
               ? '0' // Reset
-              : ethers.constants.MaxUint256, // Infinite approve
+              : ethers.MaxUint256, // Infinite approve
           );
 
         const walletAddress = await this.signer.getAddress();
         const nonce = await this.provider.getTransactionCount(walletAddress, 'pending');
-        const gasPrice = await this.provider.getGasPrice();
+        const gasPrice = unsignedApproveTx.gasPrice;
```

Comment:
Make sure that gasPrice in `unsignedApproveTx.gasPrice` is not null / undefined (what is source of gas price in `unsignedApproveTx`?)